File: src/main/java/site/gachontable/gachontablebe/global/biztalk/sendBiztalk.java
package site.gachontable.gachontablebe.global.biztalk;

import lombok.extern.slf4j.Slf4j;
import net.nurigo.sdk.NurigoApp;
import net.nurigo.sdk.message.model.KakaoOption;
import net.nurigo.sdk.message.model.Message;
import net.nurigo.sdk.message.request.SingleMessageSendingRequest;
import net.nurigo.sdk.message.response.SingleMessageSentResponse;
import net.nurigo.sdk.message.service.DefaultMessageService;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import site.gachontable.gachontablebe.global.biztalk.exception.BiztalkSendFailException;

import java.net.HttpURLConnection;
import java.util.HashMap;
import java.util.Objects;

@Slf4j
@Configuration
public class sendBiztalk {

    private final DefaultMessageService defaultMessageService;
    private final String pfId;
    private final String senderPhoneNumber;

    public sendBiztalk(
            @Value("${biztalk.app_key}") String appkey,
            @Value("${biztalk.secret_key}") String secretKey,
            @Value("${biztalk.pfId}") String pfId,
            @Value("${biztalk.sender_phoneNumber}") String senderPhoneNumber
    ) {
        this.defaultMessageService = NurigoApp.INSTANCE.initialize(appkey, secretKey, "https://api.solapi.com");
        this.pfId = pfId;
        this.senderPhoneNumber = senderPhoneNumber;
    }

    public void execute(String templateId, String phoneNumber, HashMap<String, String> variables) {
        KakaoOption kakaoOption = new KakaoOption();
        kakaoOption.setDisableSms(true);

        kakaoOption.setPfId(pfId);
        kakaoOption.setTemplateId(templateId);
        kakaoOption.setVariables(variables);

        Message message = new Message();
        message.setFrom(senderPhoneNumber);
        message.setTo(phoneNumber);
        message.setKakaoOptions(kakaoOption);

        SingleMessageSentResponse response = defaultMessageService.sendOne(new SingleMessageSendingRequest(message));

        if (!Objects.equals(response.getStatusCode(), String.valueOf(HttpURLConnection.HTTP_OK))) {
            log.error("알림톡 전송 실패. 코드: {}, 메시지: {}", response.getStatusCode(), response.getStatusMessage());
            throw new BiztalkSendFailException();
        }
    }
}

File: src/main/kotlin/site/gachontable/infra/biztalk/SendBiztalk.kt
package site.gachontable.infra.biztalk

import net.nurigo.sdk.NurigoApp.initialize
import net.nurigo.sdk.message.model.KakaoOption
import net.nurigo.sdk.message.model.Message
import net.nurigo.sdk.message.request.SingleMessageSendingRequest
import net.nurigo.sdk.message.service.DefaultMessageService
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration
import site.gachontable.infra.biztalk.exception.BiztalkSendFailException

@Configuration
class SendBiztalk(
    @Value("\${biztalk.app_key}")
    private val appKey: String,

    @Value("\${biztalk.secret_key}")
    private val secretKey: String,

    @Value("\${biztalk.pfId}")
    private val pfId: String,

    @Value("\${biztalk.sender_phoneNumber}")
    private val senderPhoneNumber: String,
) {
    private val defaultMessageService: DefaultMessageService =
        initialize(appKey, secretKey, "https://api.solapi.com")

    fun execute(templateId: String, phoneNumber: String, variables: HashMap<String, String>) {
        val kakaoOption = KakaoOption().apply {
            disableSms = true
            this.pfId = this@SendBiztalk.pfId
            this.templateId = templateId
            this.variables = variables
        }

        val message = Message().apply {
            from = senderPhoneNumber
            to = phoneNumber
            kakaoOptions = kakaoOption
        }

        val response = defaultMessageService.sendOne(SingleMessageSendingRequest(message))

        if (response!!.statusCode != "2000") {
            log.error("알림톡 전송 실패. 코드: {}, 메시지: {}", response.statusCode, response.statusMessage)
            throw BiztalkSendFailException()
        }
    }

    companion object {
        private val log = LoggerFactory.getLogger(SendBiztalk::class.java)
    }
}
